drivers/SmartThings/matter-appliance/src/matter-dishwasher/init.lua

```diff
@@ -0,0 +1,316 @@
+      if dt.device_type_id == DISHWASHER_DEVICE_TYPE_ID then
+        return true
+      end
+    end
+  end
+  return false
+end
+
+local function selected_temperature_level_attr_handler(driver, device, ib, response)
+  local tl_eps = embedded_cluster_utils.get_endpoints(device, clusters.TemperatureControl.ID, {feature_bitmap = clusters.TemperatureControl.types.Feature.TEMPERATURE_LEVEL})
+  if #tl_eps == 0 then
+    log.warn_with({ hub_logs = true }, string.format("Device does not support TEMPERATURE_LEVEL feature"))
+    return
+  end
+  log.info_with({ hub_logs = true },
+    string.format("selected_temperature_level_attr_handler: %s", ib.data.value))
+
+  local temperatureLevel = ib.data.value
+  for i, tempLevel in ipairs(supportedTemperatureLevels) do
+    if i - 1 == temperatureLevel then
+      device:emit_event_for_endpoint(ib.endpoint_id, capabilities.temperatureLevel.temperatureLevel(tempLevel))
+      break
+    end
+  end
+end
+
+local function supported_temperature_levels_attr_handler(driver, device, ib, response)
+  local tl_eps = embedded_cluster_utils.get_endpoints(device, clusters.TemperatureControl.ID, {feature_bitmap = clusters.TemperatureControl.types.Feature.TEMPERATURE_LEVEL})
+  if #tl_eps == 0 then
+    log.warn_with({ hub_logs = true }, string.format("Device does not support TEMPERATURE_LEVEL feature"))
+    return
+  end
+  log.info_with({ hub_logs = true },
+    string.format("supported_temperature_levels_attr_handler: %s", ib.data.elements))
+
+  supportedTemperatureLevels = {}
+  for _, tempLevel in ipairs(ib.data.elements) do
+    table.insert(supportedTemperatureLevels, tempLevel.value)
+  end
+  device:emit_event_for_endpoint(ib.endpoint_id, capabilities.temperatureLevel.supportedTemperatureLevels(supportedTemperatureLevels))
+end
+
+local function dishwasher_supported_modes_attr_handler(driver, device, ib, response)
+  dishwasherModeSupportedModes = {}
+  for _, mode in ipairs(ib.data.elements) do
+    if version.api < 10 then
+      clusters.DishwasherMode.types.ModeOptionStruct:augment_type(mode)
+    end
+    table.insert(dishwasherModeSupportedModes, mode.elements.label.value)
+  end
+  device:emit_event_for_endpoint(ib.endpoint_id, capabilities.mode.supportedModes(dishwasherModeSupportedModes))
+end
+
+local function dishwasher_mode_attr_handler(driver, device, ib, response)
+  log.info_with({ hub_logs = true },
+    string.format("dishwasher_mode_attr_handler mode: %s", ib.data.value))
+
+  local currentMode = ib.data.value
+  for i, mode in ipairs(dishwasherModeSupportedModes) do
+    if i - 1 == currentMode then
+      device:emit_event_for_endpoint(ib.endpoint_id, capabilities.mode.mode(mode))
+      break
+    end
+  end
+end
+
+local function dishwasher_alarm_attr_handler(driver, device, ib, response)
+  log.info_with({ hub_logs = true },
+    string.format("dishwasher_alarm_attr_handler state: %s", ib.data.value))
+
+  local isWaterFlowRateAlarm = false
+  local isContacSensorAlarm = false
```

Comment:
typo: `Contact`

Reply:
done in [39cdb6f](https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/pull/1422/commits/39cdb6f5aa61b8bcd2614de83c774e438afa8a3d)